### PR TITLE
Add Beef language

### PR DESCRIPTION
**Pull request by Nixinova**
Adds the Beef language, extension `.bf`.

Implements #5066

Usage seems to check out: https://github.com/github/linguist/issues/5066#issuecomment-774380976

<del>Color `#4BB4E3` averaged from background of https://beeflang.org</del>
<ins>Color `#A52F4E` taken from middle of [logo](https://www.beeflang.org/img/Beef256.png)</ins>

Use C# grammar as Beef is "most[ly] directly derived from C#" - https://github.com/beefytech/Beef

All samples from beefytech/Beef licensed under MIT:
1. [Program.bf](https://raw.githubusercontent.com/beefytech/Beef/5d5186f8127af4c670f76d7af7df735d3544a16d/IDE/dist/lib/gen/src/Program.bf)
2. [RandoCode.bf](https://github.com/beefytech/Beef/blob/master/BeefTools/RandoCode/src/Program.bf)
3. [ProfilePanel.bf](https://github.com/beefytech/Beef/blob/a80410786d478f36d18b312e95589c4ece7cca4f/BeefTools/BeefPerf/src/ProfilePanel.bf)

## Checklist
- [x] Added language
- [ ] Implemented heuristic
- [x] Added sample
- [x] Added grammar (C#)

### CODE REVIEW

**Comment by Nixinova** [comment]
Brainf— and HyPhy don't seem to be in `heuristics.yml` or `heuristics.rb`; how do I add the Beef heuristic, and how are the others disambiguated?

**Comment by Alhadis** [comment]
> how do I add the Beef heuristic, and how are the others disambiguated?

Heuristics are optional. They're only necessary when our classifier does a poor job of telling two languages apart. Brainfuck's syntax is—to the surprise of absolutely nobody—easy to distinguish from HyPhy. Hence our classifier gets it right based off of our samples.

Usually, a heuristic matches a predictable and repetitive pattern, or syntax that's valid in one language but illegal in another. If neither is possible, it's best to avoid heuristics altogether and leave it to our classifier to figure it out.

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -451,6 +451,16 @@ Batchfile:
   ace_mode: batchfile
   color: "#C1F12E"
   language_id: 29
+Beef:
+  type: programming
+  color: "#4BB4E3"
```
Comment (Alhadis) [inline lib/linguist/languages.yml:456]: > `#4BB4E3`

Come on, nobody likes a blue steak. 😉 We all know what the *real* colour should be:

# `#BEEEEF`

I'm kidding, of course. The colour should actually be `#a52f4e`: sampled from the centre of the [language's logo](https://www.beeflang.org/img/Beef256.png), where its most visually-prominent colour is.

**Review by Alhadis — CHANGES_REQUESTED** [review]

**Comment by lildude** [comment]
As you're planning to rely on the classifier to correctly identify Beef `.bf` files, can you please add at least one more sample. The new "Classifier cross-validation" test allows us to confirm the classifier behaviour, however it needs at least two samples:

From the test output:

```
WARNING: No cross-validation possible for Beef (only one sample)
```

**Comment by Nixinova** [comment]
Added two more

**Comment by lildude** [comment]
> Added two more

Thanks. Can you update the OP with links to the sources and their licenses too, and then I think we're good to go.

**Comment by Nixinova** [comment]
Updated

**Review by lildude — APPROVED** [review]

**Pull request merged**